packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -4,14 +4,70 @@ const chalk = require(`chalk`)
 const { formatPluginOptionsForCLI } = require(`./plugin-options`)
 const { CODES } = require(`./report`)
 
+const createContentfulErrorMessage = e => {
+  if (typeof e === `string`) {
+    return e
+  }
+
+  // If we get a response, use this as basis to create the error message
+  // Contentful JS SDK tends to give errors with different structures:
+  // https://github.com/contentful/contentful.js/blob/b67b77ac8c919c4ec39203f8cac2043854ab0014/lib/create-contentful-api.js#L89-L99
+  if (e.response) {
+    e = { ...e, ...e.response }
+
+    if (typeof e.response.data === `string`) {
```

Comment:
when would this be a string? and when an object? Why isn't it consistent?

Reply:
Error catching and logging via Contentful JS SDK is a mess. I already escalated, but a fix probably will need 1-2 years.

Reply:
![kill me now](https://media.giphy.com/media/3ORizZtUhUXFm/giphy.gif)

Reply:
@wardpeet have one more for you. See how "easy" the official import/export tools are parsing errors:

https://github.com/contentful/contentful-batch-libs/blob/master/lib/logging.js#L12-L92

I cleaned up my solution and added more comments